lib/db/schema/patch-085-086.sql

```diff
@@ -0,0 +1,45 @@
+SET NAMES utf8mb4 COLLATE utf8mb4_bin;
+
+CALL assertPatchLevel('85');
+
+ALTER TABLE recoveryKeys ADD COLUMN recoveryKeyIdHash BINARY(32),
+ALGORITHM = INPLACE, LOCK = NONE;
+
+CREATE PROCEDURE `createRecoveryKey_3` (
+  IN `uidArg` BINARY(16),
+  IN `recoveryKeyIdHashArg` BINARY(32),
+  IN `recoveryDataArg` TEXT
+)
+BEGIN
+
+  DECLARE EXIT HANDLER FOR SQLEXCEPTION
+  BEGIN
+    ROLLBACK;
+    RESIGNAL;
+  END;
+
+  START TRANSACTION;
+
+  SET @accountCount = 0;
+
+  -- Signal error if no user found
+  SELECT COUNT(*) INTO @accountCount FROM `accounts` WHERE `uid` = `uidArg`;
+  IF @accountCount = 0 THEN
+    SIGNAL SQLSTATE '45000' SET MYSQL_ERRNO = 1643, MESSAGE_TEXT = 'Can not create recovery key for unknown user.';
+  END IF;
+
+  INSERT INTO recoveryKeys (uid, recoveryKeyIdHash, recoveryData) VALUES (uidArg, recoveryKeyIdHashArg, recoveryDataArg);
```

Comment:
I'm surprised this works without also inserting a value for the `recoveryKeyId` field; what am I missing?

Reply:
Hmm, looks like the `recoveryKeyId` column is allowed to be null (it probably shouldn't have been), which means I should make `recoveryKeyIdHash` NOT NULL.